src/sql/parts/dashboard/widgets/insights/insightsWidget.component.ts

```diff
@@ -300,28 +301,15 @@ export class InsightsWidget extends DashboardWidget implements IDashboardWidget,
 			let match = filePath.match(insertValueRegex);
 			if (match && match.length > 0 && match[1] === 'workspaceRoot') {
 				filePath = filePath.replace(match[0], '');
-
-				//filePath = this.dashboardService.workspaceContextService.toResource(filePath).fsPath;
 				switch (this.workspaceContextService.getWorkbenchState()) {
 					case WorkbenchState.FOLDER:
 						filePath = this.workspaceContextService.getWorkspace().folders[0].toResource(filePath).fsPath;
 						break;
 					case WorkbenchState.WORKSPACE:
-						let filePathArray = filePath.split('/');
-						// filter out empty sections
-						filePathArray = filePathArray.filter(i => !!i);
-						let folder = this.workspaceContextService.getWorkspace().folders.find(i => i.name === filePathArray[0]);
-						if (!folder) {
-							return Promise.reject(new Error(`Could not find workspace folder ${filePathArray[0]}`));
-						}
-						// remove the folder name from the filepath
-						filePathArray.shift();
-						// rejoin the filepath after doing the work to find the right folder
-						filePath = '/' + filePathArray.join('/');
-						filePath = folder.toResource(filePath).fsPath;
-						break;
+						// Look through all the folders in the workspace use the first one that has the file we're looking for
+						let filePaths = this.workspaceContextService.getWorkspace().folders.map(f => f.toResource(filePath)).filter(p => fs.existsSync(p.fsPath));
```

Comment:
fs sync calls are very bad to use. You should use the equivalent in vs/base/node/pfs instead.